test/jruby/test_higher_javasupport.rb

```diff
@@ -1773,16 +1773,29 @@ def test_no_warnings_on_concurrent_class_loading_with_include_package
       include_package 'java.math'
       include_package 'java.net'
     end
+
+    thread_count = 50
+
     output = with_stderr_captured do
-      threads = (0..100).map do
+      latch = java.util.concurrent.CountDownLatch.new(1)
+      waiting_thread_count = java.util.concurrent.atomic.AtomicInteger.new
```

Comment:
What about making this one a CountDownLatch too? Waiting for something to happen _N_ times is what they're designed for, even if I do often end up using them with _N_ = 1 😄